lib/aws/rails/action_mailbox/sns_notification.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-sns'
+require 'aws/rails/action_mailbox/s3_client'
+
+module Aws
+  module Rails
+    module ActionMailbox
+      # @api private
+      class SnsNotification
+        class MessageContentError < StandardError; end
+
+        def initialize(request_body)
+          @request_body = request_body
+        end
+
+        def subscription_confirmed?
+          (200..299).cover?(confirmation_response.code.to_i)
+        end
+
+        def verified?
+          Aws::SNS::MessageVerifier.new.authentic?(@request_body)
```

Comment:
The MessageVerifier will cache pems - we may want to memoize it (similar to what was done for the S3 client) to take advantage of that and avoid downloading the same cert repeatedly.